precompiles/dapps-staking/DappsStaking.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+pragma solidity >=0.7.0;
+
+/// Interface to the precompiled contract on Shibuya/Shiden/Astar
+/// Predeployed at the address 0x0000000000000000000000000000000000005001
+interface DappsStaking {
+
+    // Storage getters
+
+    /// @dev Read current era.
+    /// @return The current era
+    function read_current_era() external view returns (uint256);
+
+    /// @dev Read unbonding period constant.
+    /// @return The unbonding period
+    function read_unbonding_period() external view returns (uint256);
+
+    /// @dev Read Total network reward for the given era
+    /// @return Total network reward for the given era
+    function read_era_reward(uint32 era) external view returns (uint128);
+
+    /// @dev Read Total staked amount for the given era
+    /// @return Total staked amount for the given era
+    function read_era_staked(uint32 era) external view returns (uint128);
+
+    /// @dev Read Staked amount for the staker
+    /// @return Staked amount for the staker
+    function read_staked_amount(address staker) external view returns (uint128);
+
+    /// @dev Read the amount staked on contract in the given era
+    /// @return The amount staked on contract in the given era
+    function read_contract_era_stake(address contract_id, uint32 era) external view returns (uint128);
```

Comment:
@Dinonard not sure what would be use of this method if era is removed. The procompile is calling `Pallet::<R>::staking_info(&contract_id, era)` 
Without era there is not much to call. Do you mean we then call latest stored/changed staking info?

Reply:
Yes, we'd just return the latest value.

The reasoning behind is that I'm not yet sure how our storage schema will evolve in the future.
Consider the current way of storing this piece of information - we have a _sparse array_ in a sense. This storage will grow forever if we never clean old history - each era in which contract stake is modified, a new entry is created in the DB.

Now, if we decide to clean the history (and I'm pretty sure we'll have to do that because unbounded DB growth means problems), you would be returning false information from this method when `era` is very old.

Imo, adding method that returns **latest** staked value should be enough - in the future, when pallet matures more, we can extend this interface with a new method that takes eras.

Reply:
implemented without era